File: Store/src/Sales.Domain/Model/Discounts/PercentageBasedDiscount.cs
﻿using System;

namespace Sales.Domain.Model.Discounts
{
    public class PercentageBasedDiscount : DiscountStrategy
    {
        private readonly float _percent;
        public PercentageBasedDiscount(float percent)
        {
            _percent = percent;
        }

        public override long CalculateDiscount(long totalPrice)
        {
            throw new NotImplementedException();   
        }
    }
}

File: src/main/java/store/sales/domain/model/discounts/PercentageBasedDiscount.java
package store.sales.domain.model.discounts;

import static java.lang.Math.round;

public class PercentageBasedDiscount extends DiscountStrategy {

    private final int percent;

    public PercentageBasedDiscount(int percent) {
        this.percent = percent;
    }

    @Override
    public int calculateDiscount(int totalPrice) {
        var discount = percent * totalPrice / 100;
        return round(discount);
    }
}
